File: 04_16/B.cpp
#include <iostream>
using namespace std;

int n, m, l, r;
 
int main (){
    int ones = 0;
    int c;
	cin >> n >> m;
	for (int i = 0; i < n; i++){
        cin >> c;
        ones +=(c==1);
	}
    ones=min(ones,n-ones);
	for (int i = 0; i < m; i++){
		cin >> l >> r;
		c = (((r - l) + 1)%2) && ((r-l)/2< ones);
        cout<<c<<endl;
	}
}

File: 04_16/B.py
if __name__ == '__main__':
    n, m = map(int, input().split())
    ones = input().count('-1')
    x = min(n - ones, ones) * 2
    for i in range(m):
        l, r = map(int, input().split())
        c = int(((r - l) % 2) and ((r - l) < x))
        print(c)
